cmd/up/space/billing/get.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 Upbound Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package billing
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	_ "embed"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"time"
+
+	gcs "cloud.google.com/go/storage"
+	"github.com/alecthomas/kong"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	gcpopt "google.golang.org/api/option"
+
+	"github.com/upbound/up/internal/usage"
+	"github.com/upbound/up/internal/usage/report"
+	reporttar "github.com/upbound/up/internal/usage/report/file/tar"
+	reportgcs "github.com/upbound/up/internal/usage/report/gcs"
+)
+
+const (
+	providerAWS   = "aws"
+	providerGCP   = "gcp"
+	providerAzure = "azure"
+
+	errFmtProviderNotSupported = "\"%s\" is not supported"
+)
+
+type dateRange usage.TimeRange
+
+func (d *dateRange) Decode(ctx *kong.DecodeContext) error {
+	var value string
+	if err := ctx.Scan.PopValueInto("date range", &value); err != nil {
+		return err
+	}
+
+	parts := strings.SplitN(value, "/", 2)
+	if len(parts) != 2 {
+		fmt.Printf("%s\n", parts)
+		return fmt.Errorf("invalid format")
+	}
+
+	start, err := time.Parse(time.DateOnly, parts[0])
+	if err != nil {
+		return err
+	}
+	d.Start = start
+
+	end, err := time.Parse(time.DateOnly, parts[1])
+	if err != nil {
+		return err
+	}
+	d.End = end
+
+	return nil
+}
+
+type provider string
+
+func (p provider) Validate() error {
+	// TODO(branden): Add support for these providers.
+	switch p {
+	case providerAWS:
+		return fmt.Errorf(errFmtProviderNotSupported, p)
+	case providerAzure:
+		return fmt.Errorf(errFmtProviderNotSupported, p)
```

Comment:
This is a bit more idiomatic:

```suggestion
	case providerGCP:
	    return nil
	default:
		return fmt.Errorf(errFmtProviderNotSupported, p)
```